authors/apps/profiles/views.py

```diff
@@ -86,3 +88,44 @@ def get(self, request, **kwargs):
         profiles = Response({'profiles': serializer.data},
                             status=status.HTTP_200_OK)
         return profiles
+
+
+class ProfileFollowAPIView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ProfileJSONRenderer,)
+    serializer_class = ProfileSerializer
+
+    def follow_unfollow(self, username, request, check, status_):
+        follower = self.request.user.userprofile
+        try:
+            followee = UserProfile.objects.get(user__username=username)
+        except UserProfile.DoesNotExist:
+            return Response(
+                {'Message': 'No profile with this username was found'},
+                status=status.HTTP_404_NOT_FOUND
+            )
+
+        if follower.pk is followee.pk:
+            return Response({
+                'Message': 'You can only follow others, not yourself'
+            }, status=status.HTTP_409_CONFLICT)
```

Comment:
I love that you took the effort to implement one cannopt follow themselves. Nice attention to detail.👌🏾